src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1933,15 +1951,30 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 }
             }
 
-            tgtPrefUse = BuildUse(embOp2Node->Op(1));
-            srcCount += 1;
-
-            for (size_t argNum = 2; argNum <= numArgs; argNum++)
+            switch (intrinEmb.id)
             {
-                srcCount += BuildDelayFreeUses(embOp2Node->Op(argNum), embOp2Node->Op(1));
-            }
+                // Special-case, CreateBreakPropagateMask's op2 is the RMW node.
+                case NI_Sve_CreateBreakPropagateMask:
+                    assert(tgtPrefEmbOp2OfOp2);
+                    assert(intrin.op3->isContained());
+                    assert(intrin.op3->IsVectorZero());
+                    tgtPrefUse = BuildUse(embOp2Node->Op(2));
+                    srcCount += 1;
+                    srcCount += BuildDelayFreeUses(embOp2Node->Op(1), embOp2Node->Op(2));
```

Comment:
I am little worried about the handling of this. Can you share the JitDump of one of the simpler repro scenario?

Reply:
Sure, I'll get a JitDump of the LoadBasicScenario.

What specifically are you worried about?

Reply:
[dump.txt](https://github.com/user-attachments/files/16237338/dump.txt)

Reply:
here we are not building refpositions in order. Basically it should be for `Op(1)`, `Op(2)` and then `Op(3)`, but here, we are doing `Op(2)`, `Op(1)` and `Op(3)`.

Reply:
i have fixed this properly and all tests are passing.